sunpy/io/ana.py

```diff
@@ -44,6 +45,9 @@ def read(filename, debug=False, **kwargs):
     --------
     >>> data = sunpy.io.ana.read(filename)  # doctest: +SKIP
     """
+    if isinstance(filename, io.IOBase):
+        filename = filename.name
```

Comment:
If a reader can't handle this, then we should not support it.

Reply:
How about raising a warning like,
`warning: reader doesn't support file-obj, falling back to file path.`(or something similar), rather than a exception?

Reply:
I am not sure, maybe a warning is enough.